### PR TITLE
[Placement Group]Enhance create placement group java api

### PR DESCRIPTION
**Pull request by clay4megtr**
## Why are these changes needed?
Enhance create placement group java api, include: 
1. user can specify pacrment group name, default name is 'unnamed_group'
2. enhance `PlacementGroupImpl `, like `name ` `bundles ` `strategy ` `PlacementGroupState` .

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: java/runtime/src/main/java/io/ray/runtime/placementgroup/PlacementGroupImpl.java [hunk 1]
```diff
@@ -1,28 +1,115 @@
...
+  private final String name;
+  private final List<Map<String, Double>> bundles;
+  private final PlacementStrategy strategy;
+  private final PlacementGroupState state;
 
-  public PlacementGroupImpl() {
-  }
-
-  public PlacementGroupImpl(PlacementGroupId id, int bundleCount) {
+  public PlacementGroupImpl(PlacementGroupId id, String name,
```
Comment (ffbin) [inline java/runtime/src/main/java/io/ray/runtime/placementgroup/PlacementGroupImpl.java:19]: private?

File: java/api/src/main/java/io/ray/api/Ray.java [hunk 2]
```diff
@@ -245,13 +245,19 @@ public static RuntimeContext getRuntimeContext() {
...
    */
+  public static PlacementGroup createPlacementGroup(String name,
+      List<Map<String, Double>> bundles, PlacementStrategy strategy) {
+    return internal().createPlacementGroup(name, bundles, strategy);
+  }
+
   public static PlacementGroup createPlacementGroup(List<Map<String, Double>> bundles,
       PlacementStrategy strategy) {
-    return internal().createPlacementGroup(bundles, strategy);
+    return createPlacementGroup("unnamed_group", bundles, strategy);
```
Comment (ffbin) [inline java/api/src/main/java/io/ray/api/Ray.java:260]: How about define a constant("unnamed_group")?

File: java/runtime/src/main/java/io/ray/runtime/placementgroup/PlacementGroupImpl.java (hunk 1 above)
Comment (clay4444) [reply java/runtime/src/main/java/io/ray/runtime/placementgroup/PlacementGroupImpl.java:19]: Sington?  why?

File: java/api/src/main/java/io/ray/api/Ray.java (hunk 2 above)
Comment (clay4444) [reply java/api/src/main/java/io/ray/api/Ray.java:260]: ok.

Comment (clay4444) [reply java/api/src/main/java/io/ray/api/Ray.java]: updated.

**Review by ffbin — APPROVED** [review]

File: java/api/src/main/java/io/ray/api/Ray.java [hunk 3]
```diff
@@ -17,6 +17,7 @@
 public final class Ray extends RayCall {
 
   private static RayRuntime runtime = null;
+  public static final String DEFAULT_PLACEMENT_GROUP_NAME = "unnamed_group";
```
Comment (kfstorm) [inline java/api/src/main/java/io/ray/api/Ray.java:20]: Better to move it into `AbstractRayRuntime`.

**Review by kfstorm — COMMENTED** [review]
Is the name of a placement group unique? If so, I suggest we use an empty string as the default name.

**Comment by clay4444** [comment]
> Is the name of a placement group unique? If so, I suggest we use an empty string as the default name.

no, the name of a placement group is not unique, keep same with python implements.

File: java/api/src/main/java/io/ray/api/Ray.java (hunk 3 above)
Comment (clay4444) [reply java/api/src/main/java/io/ray/api/Ray.java]: updated.

File: java/runtime/src/main/java/io/ray/runtime/AbstractRayRuntime.java [hunk 4]
```diff
@@ -51,6 +51,7 @@
 
   private static final Logger LOGGER = LoggerFactory.getLogger(AbstractRayRuntime.class);
   public static final String PYTHON_INIT_METHOD_NAME = "__init__";
+  public static final String DEFAULT_PLACEMENT_GROUP_NAME = "unnamed_group";
```
Comment (kfstorm) [inline java/runtime/src/main/java/io/ray/runtime/AbstractRayRuntime.java:54]: There's no need to make it public, right?

Comment (clay4444) [reply java/runtime/src/main/java/io/ray/runtime/AbstractRayRuntime.java:54]: yes,

**Review by kfstorm — APPROVED** [review]

**Pull request merged**